### PR TITLE
Update README with more description and links.

### PR DESCRIPTION
**Pull request by tomwilkie**
- Explain a bit more what Cortex is useful for.
- Move developer details to contributor guide.
- Link to new docs.
- Consolidate all the further reading into the readme, add a bunch more.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,3 +15,70 @@ git commit -s -m "Here is my signed commit"
...
+```
+make test
+```
+
+## Playing in `minikube`
+
+First, start `minikube`.
+
+You may need to load the Docker images into your minikube environment. There is
+a convenience rule in the Makefile to do this:
```
Comment (jtlisi) [inline CONTRIBUTING.md]:
```suggestion
a convenient rule in the Makefile to do this:
```

File: README.md [hunk 2]
```diff
@@ -1,123 +1,75 @@
...
-[Prometheus](https://prometheus.io) metrics when used as a [remote
-write](https://prometheus.io/docs/operating/configuration/#remote_write)
-destination, and a horizontally scalable, Prometheus-compatible query
-API.
-
-Multi-tenant means it handles metrics from multiple independent
-Prometheus sources, keeping them separate.
-
-[Instrumenting Your App: Best Practices](https://www.weave.works/docs/cloud/latest/tasks/monitor/best-instrumenting/)
+# Cortex: horizontally scalable, highly available, multi-tenant, long term storage for  Prometheus.
```
Comment (jtlisi) [inline README.md]:
```suggestion
# Cortex: horizontally scalable, highly available, multi-tenant, long term storage for Prometheus.
```

File: README.md [hunk 3]
```diff
@@ -1,123 +1,75 @@
...
-on-demand:
-
-### Weave Cloud
-
-[Weave Cloud](https://cloud.weave.works) from
-[Weaveworks](https://weave.works) lets you deploy, manage, and monitor
-container-based applications. Sign up at https://cloud.weave.works
-and follow the instructions there. Additional help can also be found
-in the [Weave Cloud documentation](https://www.weave.works/docs/cloud/latest/overview/).
+- **Horizontally scalable:** Cortex can run across multiple machines in a cluster, exceeding the throughput and storage of a single machine.  This enables you to send the metrics from multiple Prometheus servers to a single Cortex cluster and run "globally aggregated" queries across all data in a single place.
```
Comment (jtlisi) [inline README.md]:
```suggestion
- **Horizontally scalable:** Cortex can run across multiple machines in a cluster, exceeding the throughput and storage of a single machine. This enables you to send the metrics from multiple Prometheus servers to a single Cortex cluster and run "globally aggregated" queries across all data in a single place.
```
extra space

File: README.md [hunk 4]
```diff
@@ -1,123 +1,75 @@
...
+- **Highly available:** When run in a cluster, Cortex can replicate data between machines.  This allows you to survive machine failure without gaps in your graphs.
```
Comment (jtlisi) [inline README.md]:
```suggestion
- **Highly available:** When run in a cluster, Cortex can replicate data between machines. This allows you to survive machine failure without gaps in your graphs.
```
Extra space

File: docs/architecture.md [hunk 5]
```diff
@@ -128,16 +128,4 @@ The interface works somewhat differently across the supported databases:
...
-- "Weave Cortex: Multi-tenant, horizontally scalable Prometheus as a Service" (KubeCon Prometheus Day talk) [slides](http://www.slideshare.net/weaveworks/weave-cortex-multitenant-horizontally-scalable-prometheus-as-a-service) [video](https://www.youtube.com/watch?v=9Uctgnazfwk)
-- "[Cortex: Prometheus as a Service, One Year On](https://promcon.io/2017-munich/talks/cortex-prometheus-as-a-service-one-year-on/)" (PromCon 2017 Talk)
-- "Horizontally Scalable, Multi-tenant Prometheus" [slides](https://docs.google.com/presentation/d/190oIFgujktVYxWZLhLYN4q8p9dtQYoe4sxHgn4deBSI/edit#slide=id.g3b8e2d6f7e_0_6) (CNCF TOC Presentation)
-- [Cortex Query Woes](https://docs.google.com/document/d/1lsvSkv0tiAMPQv-V8vI2LZ8f4i9JuTRsuPI_i-XcAqY) provides details on Cortex query optimizations.
-- [Cortex: Horizontally Scalable, Highly Available Prometheus Monitoring](https://speakerdeck.com/grafana/cortex-horizontally-scalable-highly-available-prometheus-monitoring) (Presentation at Cloud Native London Meetup, Nov 2018)
+> The current schema recommendation is the **v6 schema**.
```
Comment (khaines) [inline docs/architecture.md]: is this line still accurate?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -15,3 +15,70 @@ git commit -s -m "Here is my signed commit"
...
+# Pick a specific version.
+go get example.com/some/module/pkg@vX.Y.Z
+```
+
+Tidy up the `go.mod` and `go.sum` files:
+
+```bash
+go mod tidy
+git add go.mod go.sum
+git commit
```
Comment (gouthamve) [inline CONTRIBUTING.md]: We also need to update vendor using `go mod vendor` and then add that too.

**Pull request merged**